logstash-core/lib/logstash/runner.rb

```diff
@@ -305,6 +305,9 @@ def execute
     if setting("config.debug") && !logger.debug?
       logger.warn("--config.debug was specified, but log.level was not set to \'debug\'! No config info will be logged.")
     end
+    if setting("pipeline.buffer.type") == "heap"
```

Comment:
should we ensure that `noPreferDirect` is disabled if buffer.type is set as "direct" (or not set)?

Reply:
Good point, we should check if the user configured the setting `pipeline.buffer.type`, then act accordingly in `configure_pipeline_buffer_type_to_heap`, which at this point should be renamed `configure_pipeline_buffer_type`.
If the user has already defined `io.netty.noPreferDirect` in his `jvm.options` **and** also defined `pipeline.buffer.type` setting then would be a configuration error.
If the user defined `pipeline.buffer.type == direct` then will set `io.netty.noPreferDirect` to false.